easybuild/easyblocks/n/neuron.py

```diff
@@ -60,85 +62,135 @@ def extra_options():
-                raise EasyBuildError("Failed to change back to %s: %s", pwd, err)
+                try:
+                    os.chdir(pwd)
+                except OSError as err:
+                    raise EasyBuildError("Failed to change back to %s: %s", pwd, err)
 
     def sanity_check_step(self):
         """Custom sanity check for NEURON."""
         shlib_ext = get_shared_lib_ext()
         binpath = os.path.join(self.hostcpu, 'bin')
         libpath = os.path.join(self.hostcpu, 'lib', 'lib%s.' + shlib_ext)
-        binaries = ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
-                    "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
-        binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc", "oc_makefile",
-                                           "ocmodl"]]
-        libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython", "oc",
-                "ocxt", "scopmath", "sparse13", "sundials"]
-
         # hoc_ed is not included in the sources of 7.4. However, it is included in the binary distribution.
         # Nevertheless, the binary has a date old enough (June 2014, instead of November 2015 like all the
         # others) to be considered a mistake in the distribution
+        binaries = []
         if LooseVersion(self.version) < LooseVersion('7.4'):
             binaries.append("hoc_ed")
 
-        custom_paths = {
+        sanity_check_dirs = ['share/nrn']
+        if LooseVersion(self.version) < LooseVersion('7.8.1'):
+            binaries += ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
+                         "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
+            binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc",
+                                               "oc_makefile", "ocmodl"]]
+            libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython",
+                    "oc", "ocxt", "scopmath", "sparse13", "sundials"]
+            sanity_check_dirs += ['include/nrn']
+        # list of included binaries changed with cmake. See
+        # https://github.com/neuronsimulator/nrn/issues/899
+        else:
+            binaries += ["mkthreadsafe", "modlunit", "neurondemo", "nocmodl", "nrngui", "nrniv", "nrnivmodl",
+                         "nrnmech_makefile", "nrnpyenv.sh", "set_nrnpyenv.sh", "sortspike"]
+            libs = ["nrniv", "rxdmath"]
+            sanity_check_dirs += ['include']
+            if self.with_python:
+                sanity_check_dirs += [os.path.join("lib", "python"),
+                                      os.path.join("lib", "python%(pyshortver)s", "site-packages")]
+
+        # (we can not pass this via custom_paths, since then the %(pyshortver)s template value will not be resolved)
+        self.cfg['sanity_check_paths'] = {
```

Comment:
This is not OK, since you're blindly overwriting `sanity_check_paths` if it's set in the easyconfig file.

But if you use `self.pyliddir` as suggested above, you can just pass `custom_paths` below as it was before.

Reply:
pylibdir results in checking the same version of python 3 times in a row 
```
== 2021-01-08 16:04:44,144 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:04:59,627 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:05:11,236 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
```